docs/index.rst

```diff
@@ -1,10 +1,68 @@
-.. Cogent documentation master file, created by
-   sphinx-quickstart on Tue Dec 17 18:14:44 2019.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
+========================================================================
+                                 Cogent
+========================================================================
+
+:Version:	2.9
+:License:	GPL-2.0-only
+:Homepage:	<https://ts.data61.csiro.au/projects/TS/cogent.pml>
+:Repository:	<https://github.com/NICTA/cogent>
 
 Welcome to Cogent's documentation!
-==================================
+
+Cogent is a restricted, polymorphic,
+higher-order and purely functional language
+with linear types
+and without the need for a trusted runtime
+or garbage collector.
+
+**Cogent is restricted.**
+Cogent is *not* a general-purpose language.
+It omits certain features
+that one might come to expect
+from a programming language.
+For instance, Cogent doesn't support general recursion.
+
+**Cogent is polymorphic.**
+Parametrically.
+This is similar to templates in C++,
+or generics in Rust.
+
+**Cogent is higher-order.**
+Functions are values --- they can be function arguments.
+
+**Cogent is purely functional,**
+in the sense of
+the functional programming paradigm:
+functions are stateless,
+and the result of a function is deterministically defined
+by the arguments to the function,
+just like a mathematical function.
+
+**Cogent has linear types**:
+as a means to control references,
+every linear object
+(roughly equivalent to heap-allocated object)
+can be referenced exactly once.
+
+**Cogent has no trusted runtime.**
+Cogent compiler generates C code,
+so the trusted component is limited to
+the C runtime system.
+
+**Cogent has no garbage collector.**
+Since Cogent compiler generates C code,
+we would manage memory manually,
+like we do in any C program.
+
```

Comment:
Shall we mention here that Cogent is a certifying compiler? The aim is to produce proofs and that's why the language is made so restricted.